lib/encode.js

```diff
@@ -13,39 +21,61 @@ import {inspect} from './util.js';
  *   CBOR-LD bytes.
  * @param {documentLoaderFunction} options.documentLoader -The document loader
  *   to use when resolving JSON-LD Context URLs.
- * @param {boolean} [options.compressionMode=1] - `1` to use compression mode
- *   version 1, `0` to use no compression.
- * @param {Map} [options.appContextMap] - A map of JSON-LD Context URLs and
- *   their encoded CBOR-LD values (must be values greater than 32767 (0x7FFF)).
- * @param {diagnosticFunction} [options.diagnose] - A function that, if
- * provided, is called with diagnostic information.
- *
+ * @param {number} options.varintValue - The varint value for the
+ *  registry entry associated with the resulting CBOR-LD payload. For legacy
+ *  support, use varintTagValue = -1.
+ * @param {Map} options.keywordsTable -A map of JSON-LD keywords
+ *  and their associated CBOR-LD integer values.
+ * @param {Map} options.stringTable - A map of string values and
+ *  their associated CBOR-LD integer values.
+ * @param {Map} options.urlSchemeTable - A map of URL schemes to
+ *  their associated CBOR-LD integer values.
+ * @param {Map} options.typedLiteralTable - A map of JSON-LD types
+ *  to maps that map values of that type to their associated CBOR-LD
+ *  integer values.
```

Comment:
```suggestion
 * @param {Map} [options.keywordsTable] - A map of JSON-LD keywords
 *  and their associated CBOR-LD integer values; for non-legacy mode only.
 * @param {Map} [options.stringTable] - A map of string values and
 *  their associated CBOR-LD integer values; for non-legacy mode only.
 * @param {Map} [options.urlSchemeTable] - A map of URL schemes to
 *  their associated CBOR-LD integer values; for non-legacy mode only.
 * @param {Map} [options.typedLiteralTable] - A map of JSON-LD types
 *  to maps that map values of that type to their associated CBOR-LD
 *  integer values; for non-legacy mode only.
```

But, even with these fixes, I'm thinking we probably need to put these instead of a `tables` object to clean up this API before we release. I think it should look like:

```js
tables: {
  keywords,
  strings,
  urlSchemes,
  typedLiterals
}
```

This would also make the public API surface consistently plural. We can keep the internals as they are or change those too if we want, whatever is quicker, but in order to do a release, I do think we should do the above and not have these non-namespaced/bucketed tables floating along with the other options.